### PR TITLE
rand uniform: fix likely usage

### PR DESCRIPTION
**Pull request by paulidale**
This is rarely hit, so it's not a big deal but better to be correct.

### CODE REVIEW

**Review by mspncp — APPROVED** [review]
nit: typo 'likey' -> 'likely' in commit message.

**Review by tom-cosgrove-arm — APPROVED** [review]

File: crypto/rand/rand_uniform.c [hunk 1]
```diff
@@ -85,7 +85,7 @@ uint32_t ossl_rand_uniform_uint32(OSSL_LIB_CTX *ctx, uint32_t upper, int *err)
         if (f < f2)
             return i + 1;
         /* For not all 1 bits, there is no carry so return the result */
-        if (unlikely(f != 0xffffffff))
+        if (likely(f != 0xffffffff))
```
Comment (tom-cosgrove-arm) [inline crypto/rand/rand_uniform.c:88]: were we going to change `likely()` and `unlikely()` to `ossl_*likely()` at some point?

Comment (tom-cosgrove-arm) [reply crypto/rand/rand_uniform.c:88]: (not in this PR, of course)

**Review by t8m — APPROVED** [review]

File: crypto/rand/rand_uniform.c (hunk 1 above)
Comment (t8m) [reply crypto/rand/rand_uniform.c:88]: Yes, PR welcome. Against the master branch.

**Comment by mspncp** [comment]
> nit: typo 'likey' -> 'likely' in commit message.

Can be done while merging imo.

**Comment by paulidale** [comment]
Fixed commit message typo - no code changes so no reset of timer or reviews.

File: crypto/rand/rand_uniform.c (hunk 1 above)
Comment (mspncp) [reply crypto/rand/rand_uniform.c:88]: See #22603.

**Comment by t8m** [comment]
Merged to the master and 3.2 branches. Thank you.

**Pull request closed without merging**